### PR TITLE
new test for rehashing using loaded dictionary

### PR DESCRIPTION
**Pull request by jschuecker**
When the dictionary is saved and loaded to/from h5, the hash should not change.

### CODE REVIEW

File: dicthash/test/test_dicthash.py [hunk 1]
```diff
@@ -266,3 +267,19 @@ def test_lists_array_tuples_are_equal(self):
...
+        d0 = {
+            'a': 'asd',
+            'b': 0.12,
+            'c': [3, 4, 5],
+            'd': np.array([[3, 4, 5], [3, 4, 5]]),
+            'e': True
+        }
+        hash0 = dicthash.generate_hash_from_dict(d0)
+        h5.add_to_h5('store_and_rehash_h5py.h5', {'d0': d0}, 'w')
+        d0 = h5.load_h5('store_and_rehash_h5py.h5', 'd0')
```
Comment (jakobj) [inline dicthash/test/test_dicthash.py]: to be more explicit: the loaded dictionary should be called `d1`. (also in the line below)

**Comment by jakobj** [comment]
this PR adds dependency on `h5py_wrapper`. the CI fails due to unmet dependencies. we need to adapt the travis script to get it to fetch the wrapper. this could be done directly in the .travis.yml or by creating a makefile that installs all dependencies.
does this test work locally?

ps: or maybe we can make the test conditional, i.e., it is only executed if `h5py_wrapper` is found. this would also make it easier for other people to run the tests.

**Comment by mschmidt87** [comment]
I think we should do both: Add the h5py_wrapper to the dependencies of travis and make it conditional to make it easier for people to run the tests locally.

**Comment by jakobj** [comment]
working on it. will start a PR against @jschuecker's branch once i am done.

**Comment by jakobj** [comment]
looks good :+1:

**Pull request merged**